src/runtime/virtcontainers/qemu.go

```diff
@@ -1325,6 +1335,9 @@ func (q *qemu) hotplugVFIODevice(device *config.VFIODev, op operation) (err erro
 		case config.VFIODeviceNormalType:
 			return q.qmpMonitorCh.qmp.ExecutePCIVFIODeviceAdd(q.qmpMonitorCh.ctx, devID, device.BDF, addr, bridge.ID, romFile)
 		case config.VFIODeviceMediatedType:
+			if utils.IsAPVFIOMediatedDevice(device.SysfsDev) {
+				return q.qmpMonitorCh.qmp.ExecuteAPVFIOMediatedDeviceAdd(q.qmpMonitorCh.ctx, device.SysfsDev)
+			}
```

Comment:
hold on, why I'm seeing this change twice ? :confused:

Reply:
One is inside the [`if q.state.HotplugVFIOOnRootBus` block](https://github.com/kata-containers/kata-containers/blob/dd171ec8e9650860d0250aeb9eab3e5350fae722/src/runtime/virtcontainers/qemu.go#L1290), one [outside](https://github.com/kata-containers/kata-containers/blob/dd171ec8e9650860d0250aeb9eab3e5350fae722/src/runtime/virtcontainers/qemu.go#L1339).